cmd/dp-hierarchy-builder/main.go

```diff
@@ -2,127 +2,194 @@ package main
+	case err := <-kafkaConsumer.Channels().Errors:
+		log.Event(ctx, "kafka consumer error", log.ERROR, log.Error(err))
+	case err := <-kafkaProducer.Channels().Errors:
+		log.Event(ctx, "kafka result producer error", log.ERROR, log.Error(err))
+	case err := <-kafkaErrorProducer.Channels().Errors:
+		log.Event(ctx, "kafka error producer error", log.ERROR, log.Error(err))
 	case err := <-errorChannel:
-		log.ErrorC("error channel", err, nil)
+		log.Event(ctx, "error channel error", log.ERROR, log.Error(err))
 	case <-signals:
-		log.Debug("os signal received", nil)
+		log.Event(ctx, "os signal received", log.INFO)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+	ctx, cancel := context.WithTimeout(ctx, cfg.GracefulShutdownTimeout)
 
 	// gracefully dispose resources
+	hc.Stop()
+
+	err = httpServer.Close(ctx)
+	logIfError(ctx, err, "error closing http server")
+
 	err = eventConsumer.Close(ctx)
-	logIfError(err)
+	logIfError(ctx, err, "error closing event consumer")
 
 	err = kafkaConsumer.Close(ctx)
-	logIfError(err)
+	logIfError(ctx, err, "error closing kafka consumer")
 
 	err = kafkaProducer.Close(ctx)
-	logIfError(err)
+	logIfError(ctx, err, "error closing kafka producer")
 
 	err = kafkaErrorProducer.Close(ctx)
-	logIfError(err)
-
-	err = healthChecker.Close(ctx)
-	logIfError(err)
+	logIfError(ctx, err, "error closing kafka error producer")
 
 	err = db.Close(ctx)
-	logIfError(err)
+	logIfError(ctx, err, "error closing graph db connection")
 
 	// cancel the timer in the shutdown context
 	cancel()
 
-	log.Debug("graceful shutdown was successful", nil)
+	log.Event(ctx, "graceful shutdown was successful", log.INFO)
 	os.Exit(1)
 }
 
-func exitIfError(err error) {
+func startApi(ctx context.Context, hc healthcheck.HealthCheck, cfg *config.Config) (chan error, *server.Server) {
+	router := mux.NewRouter()
+	router.HandleFunc("/health", hc.Handler)
+	apiErrors := make(chan error, 1)
+
+	httpServer := server.New(cfg.BindAddr, router)
+	// Disable this here to allow main to manage graceful shutdown of the entire app.
+	httpServer.HandleOSSignals = false
+
+	go func() {
+		log.Event(ctx, "starting api", log.INFO)
+		if err := httpServer.ListenAndServe(); err != nil {
+			log.Event(ctx, "http server error", log.ERROR, log.Error(err))
+			apiErrors <- err
+		}
+	}()
+	return apiErrors, httpServer
+}
+
+func startHealthChecks(
+	ctx context.Context,
+	cfg *config.Config,
+	kafkaConsumer *kafka.ConsumerGroup,
+	kafkaProducer *kafka.Producer,
+	kafkaErrorProducer *kafka.Producer,
+	db *graph.DB) healthcheck.HealthCheck {
+
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	exitIfError(ctx, err, "error creating version info")
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
+	err = hc.AddCheck("Kafka Consumer", kafkaConsumer.Checker)
+	exitIfError(ctx, err, "error creating kafka consumer")
```

Comment:
I would allow for each error to be logged from each addCheck so we can identify with health checkers have failed to initialise.

Then exit on line 179 if there are any health checker errors